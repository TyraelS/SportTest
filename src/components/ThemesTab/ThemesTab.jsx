import React from 'react';
import PropTypes from 'prop-types';

import ThemesTabWrapper from './ThemesTabWrapper.style';
import ThemesTabItem from 'components/ThemesTabItem';
import * as themes from 'themes';

const ThemesTab = ({ themesTabItemClick }) => (
  <ThemesTabWrapper>
    <span>Choose a theme</span>
    {Object.keys(themes).map(id => (
      <ThemesTabItem key={id} id={id} handler={themesTabItemClick} />
    ))}
  </ThemesTabWrapper>
);

ThemesTab.displayName = 'ThemesTab';
ThemesTab.propTypes = {
  themesTabItemClick: PropTypes.func.isRequired
};

export default ThemesTab;
