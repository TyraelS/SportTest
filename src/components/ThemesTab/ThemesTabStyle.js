import styled from 'styled-components';

const ThemesTab = styled.div`
  display: flex;
  width: 10%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: ${props => props.theme.regular};
  color: ${props => props.theme.textThemes};
  > * {
    padding: 15px;
  }
`;

export default ThemesTab;
