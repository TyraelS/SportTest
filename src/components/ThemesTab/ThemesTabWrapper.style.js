import styled from 'styled-components';

const ThemesTabWrapper = styled.div`
  display: flex;
  width: 10%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: ${props => props.theme.regular};
  color: ${props => props.theme.textThemes};
  > span,
  > div {
    padding: 15px;
  }
`;

export default ThemesTabWrapper;
