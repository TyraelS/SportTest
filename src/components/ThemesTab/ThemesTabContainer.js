import { compose, setDisplayName, withHandlers } from 'recompose';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { setTheme } from 'reducers/theme';
import ThemesTab from './ThemesTab';

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setTheme: setTheme
    },
    dispatch
  );

export const handleSetTheme = {
  themesTabItemClick: ({ setTheme }) => id => {
    setTheme(id);
  }
};

const enhance = compose(
  setDisplayName('ThemesTabContainer'),
  connect(
    null,
    mapDispatchToProps
  ),
  withHandlers(handleSetTheme)
);

export default enhance(ThemesTab);
