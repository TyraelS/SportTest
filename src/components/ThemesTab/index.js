import ThemesTabContainer from './ThemesTabContainer';

export default ThemesTabContainer;
