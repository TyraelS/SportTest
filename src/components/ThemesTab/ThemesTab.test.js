import React from 'react';
import { shallow } from 'enzyme';
import ThemesTab from './ThemesTab';

describe('Given the ThemesTab component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(<ThemesTab themesTabItemClick={jest.fn()} />);
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
