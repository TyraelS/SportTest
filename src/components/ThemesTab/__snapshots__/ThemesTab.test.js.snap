// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ThemesTab component when the component is rendered should match the snapshot 1`] = `
<ForwardRef(styled.div)>
  <span>
    Choose a theme
  </span>
  <ThemesTabItem
    handler={[MockFunction]}
    id="light"
    key="light"
  />
  <ThemesTabItem
    handler={[MockFunction]}
    id="dark"
    key="dark"
  />
</ForwardRef(styled.div)>
`;
